app/src/main/java/bg/zahov/app/ui/home/HomeFragment.kt

```diff
@@ -9,98 +8,30 @@ import android.view.MenuItem
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?,
     ): View {
-        _binding = FragmentHomeBinding.inflate(inflater, container, false)
-        setupTopBar()
         requireActivity().showBottomNav()
         requireActivity().showTopBar()
-        return binding.root
-    }
-
-    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        super.onViewCreated(view, savedInstanceState)
-        binding.apply {
-            homeViewModel.state.map { HomeUiMapper.map(it) }.observe(viewLifecycleOwner) {
-                loadingIndicator.visibility = it.loadingVisibility
-                weeklyWorkoutsChart.apply {
-                    visibility = it.chartVisibility
-                    xAxis.apply {
-                        valueFormatter = IndexAxisValueFormatter(it.weekRanges.toTypedArray())
-                        axisMaximum = it.xMax
-                        axisMinimum = it.xMin
-                    }
-                    axisRight.apply {
-                        axisMaximum = it.yMax
-                        axisMinimum = it.yMin
-                    }
-
-                    val dataSet = BarDataSet(it.lineData, "workouts")
-                    dataSet.setDrawValues(false)
-                    val barData = BarData(dataSet)
-                    barData.barWidth = 0.5f
-                    data = barData
-                    weeklyWorkoutsChart.notifyDataSetChanged()
-                    weeklyWorkoutsChart.invalidate()
-                }
-                numberOfWorkouts.apply {
-                    visibility = it.textFieldVisibility
-                    text = getString(R.string.workout_text, it.numberOfWorkouts)
-                }
-            }
-            homeViewModel.userName.observe(viewLifecycleOwner) {
-                profileName.text = it
-            }
-
-            weeklyWorkoutsChart.apply {
-                setFitBars(true)
-                legend.isEnabled = false
-                isDoubleTapToZoomEnabled = false
-                axisLeft.isEnabled = false
-                isDragEnabled = false
-                isHighlightFullBarEnabled = false
-                description.apply {
-                    setPosition(250f, 60f)
-                    text = "Weekly workouts"
-                    textColor = Color.WHITE
-                }
-                xAxis.apply {
-                    position = XAxis.XAxisPosition.BOTTOM
-                    granularity = 1f
-                    axisMinimum = 0f
-                    textColor = Color.WHITE
-                    setCenterAxisLabels(true)
-                    isGranularityEnabled = true
-                }
-                axisRight.apply {
-                    textColor = Color.WHITE
-                    granularity = 1f
-                    setDrawGridLines(false)
-                }
+        setupTopBar()
+        return ComposeView(requireContext()).apply {
```

Comment:
Same here. `context`

Reply:
We no longer have a xml file for this screen and we don't have a compose View however,  the ComposeView constructor requires a non nullable context and I had to use this

Reply:
Let's make a special exception only for this use case. Normally we'd want to check for null context and handle the case where it's null, but in this exact case it should be fine.